event/batch_test.go

```diff
@@ -0,0 +1,200 @@
+package event_test
+
+import (
+	"github.com/ONSdigital/dp-observation-importer/event"
+	"github.com/ONSdigital/dp-observation-importer/schema"
+	"github.com/ONSdigital/go-ns/kafka/kafkatest"
+	. "github.com/smartystreets/goconvey/convey"
+	"testing"
+)
+
+func TestIsEmpty(t *testing.T) {
+
+	Convey("Given a batch that is not empty", t, func() {
+
+		expectedEvent := getExampleEvent()
+		message := kafkatest.NewMessage([]byte(Marshal(*expectedEvent)))
+
+		batchSize := 1
+		batch := event.NewBatch(batchSize)
+
+		Convey("When the batch has no messages added", func() {
+			Convey("The batch is empty", func() {
+				So(batch.IsEmpty(), ShouldBeTrue)
+			})
+		})
+
+		Convey("When the batch has a message added", func() {
+
+			batch.Add(message)
+
+			Convey("The batch is now empty", func() {
+				So(batch.IsEmpty(), ShouldBeFalse)
+			})
+		})
+	})
+}
+
+func TestAdd(t *testing.T) {
+
+	Convey("Given a batch", t, func() {
+
+		expectedEvent := getExampleEvent()
+		message := kafkatest.NewMessage([]byte(Marshal(*expectedEvent)))
+
+		batchSize := 1
+		batch := event.NewBatch(batchSize)
+
+		Convey("When add is called with a valid message", func() {
+
+			batch.Add(message)
+
+			Convey("The batch contains the expected event.", func() {
+				So(len(batch.Events()), ShouldEqual, 1)
+				So(batch.Events()[0].Row, ShouldEqual, expectedEvent.Row)
+				So(batch.Events()[0].InstanceID, ShouldEqual, expectedEvent.InstanceID)
+			})
+		})
+	})
+}
+
+func TestCommit(t *testing.T) {
+
+	Convey("Given a batch with two valid messages", t, func() {
+
+		expectedEvent := event.ObservationExtracted{InstanceID: "123", Row: "the,row,content"}
+		expectedLastEvent := event.ObservationExtracted{InstanceID: "123", Row: "last,row,content"}
+		message := kafkatest.NewMessage([]byte(Marshal(expectedEvent)))
+		lastMessage := kafkatest.NewMessage([]byte(Marshal(expectedLastEvent)))
+
+		batchSize := 2
+		batch := event.NewBatch(batchSize)
+
+		batch.Add(message)
+		batch.Add(lastMessage)
+
+		batch.Commit()
+
+		Convey("When commit is called", func() {
+
+			Convey("The last message in the batch is committed", func() {
+				So(message.Committed(), ShouldBeFalse)
```

Comment:
Only the last message added to the batch is committed when all of the messages in the batch have been handled. When this happens I clear out the batch so it can be reused. Should I make the test descriptions clearer in some way?